app/uk/gov/hmrc/softdrinksindustrylevystub/services/SubscriptionGenerator.scala

```diff
@@ -16,167 +16,147 @@
-        SiteType.withName("Warehouse"),
-        SiteType.withName("ProductionSite"))
-    } yield siteType
-
-  private lazy val siteActionGen: Gen[SiteAction.Value] =
-    for {
-      siteAction <- Gen.oneOf(
-        SiteAction.withName("Unknown"),
-        SiteAction.withName("NewSite"),
-        SiteAction.withName("AmendSite"),
-        SiteAction.withName("CloseSite"),
-        SiteAction.withName("TransferSite"))
-    } yield siteAction
-
-  private lazy val activitiesGen: Gen[ActivityType.Value] =
-    for {
-      activityType <- Gen.oneOf(
-        ActivityType.withName("Unknown"),
-        ActivityType.withName("Producer"),
-        ActivityType.withName("Importer"),
-        ActivityType.withName("ContractPacker"))
-    } yield activityType
-
-  private lazy val producerClassificationGen: Gen[ProducerClassification.Value] =
-    for {
-      producerClassification <- Gen.oneOf(
-        ProducerClassification.withName("Unknown"),
-        ProducerClassification.withName("Large"),
-        ProducerClassification.withName("Small"))
-    } yield producerClassification
-
-  private lazy val levyDetailsGen: Gen[LevyDetails] = {
-    activitiesGen |@|                         // activities
-    Gen.boolean |@|                           // lessThanMillion
-    producerClassificationGen.sometimes |@|   // producerClassification
-    Gen.boolean |@|                           // smallProducerExemption
-    Gen.boolean |@|                           // usesCopacker
-    Gen.boolean                               // voluntarilyRegistered
-  }.map(LevyDetails.apply)
+  private lazy val producerDetailsGen: Gen[ProducerDetails] = {
+    Gen.boolean                                              |@| // produceMillionLitres
+    Gen.oneOf("1","2")                                       |@| // producerClassification
+    Gen.boolean.sometimes                                    |@| // smallProducerExemption
+    Gen.boolean.sometimes                                    |@| // useContractPacker
+    Gen.boolean.sometimes                                        // voluntarilyRegistered
+  }.map(ProducerDetails.apply)
+
+  private lazy val detailsGen: Gen[Details] = {
+    Gen.boolean                                              |@| // producer
+    producerDetailsGen.sometimes                             |@| // producerDetails
+    Gen.boolean                                              |@| // importer
+    Gen.boolean                                                  // contractPacker
+  }.map(Details.apply)
 
   private lazy val contactDetailsGen: Gen[ContactDetails] = {
-    Gen.forename() |@|                        // name
-    Gen.alphaNumStr.sometimes |@|             // positionInCompany
-    Gen.ukPhoneNumber |@|                     // telephoneNumber
-    pattern"99999 999999".gen.sometimes |@|   // mobileNumber
-    Gen.const("john.doe@somedomain.com")      // emailAddress
+    pattern"99999 999999".gen                                |@| // telephone
```

Comment:
Should we externalise this to a val?

Reply:
Yes. Done.

Reply:
This doesn't work as you need the quotes on the pattern call.